### ISSUE TITLE
[Investigation] find-unused finding a lot of false positives

### ISSUE THREAD

**Issue by JoeTypeDigital** [typedigital/eslint-config-typedigital#53]
**Context**

the newly installed tools to find unused rulesets find a lot of false positives.

**To Reproduce**

1. run npm run find-unused

**Expected behavior**

it should show rules that have no effect currently or are depricated.

**Goals**

1. find out which criteria label a rule as unused
2. find out how to stop this tool from finding false positives

**Comment by bob606** [comment]
This is the code from unused, I found it here: https://github.com/sarbbottam/eslint-find-rules/blob/master/src/lib/rule-finder.js.
Please see comments, to know how unused rules are determined. 

```
get currentRuleNames = _getCurrentNamesRules(config); // getAllRules from Config
  if (omitCore) {
    currentRuleNames = currentRuleNames.filter(_isNotCore);
  }

  const pluginRules = _getPluginRules(config); // getAllRules from plugin 
  const coreRules = _getCoreRules(); // getAllRules from core
  const allRules = omitCore ? pluginRules : new Map([...coreRules, ...pluginRules]); // combine plugin rules [and core rules]

  let allRuleNames = [...allRules.keys()]; 
  let pluginRuleNames = [...pluginRules.keys()];
  if (!includeDeprecated) {
    allRuleNames = _filterRuleNames(allRuleNames, allRules, _notDeprecated);
    pluginRuleNames = _filterRuleNames(pluginRuleNames, pluginRules, _notDeprecated);
  }
  const deprecatedRuleNames = _filterRuleNames(currentRuleNames, allRules, _isDeprecated);
  const dedupedRuleNames = [...new Set(allRuleNames)];
  const unusedRuleNames = difference(dedupedRuleNames, currentRuleNames); // check config rules against plugin rules [and core rules] and give back duplicates
```

**Comment by timidak** [comment]
So if I got it correctly, `find-unused` finds rules in the ESLint standard rules and the plugins which are not used or configured by our config. That means that we have to use each rule and have to intentionally turn it `'off'` if we don't want it to take effect. 

I run the script and saw that we have many unimplemented rules which we should implemented, so that we now what our options are and if we do not care we just turn the rules off. 

### Summary

So to summarize the tool is doing exactly what it should and does not find false positives in my opinion @JoeTypeDigital.
@bob606 the algorithm you found seems to support my thesis, or what do you mean? 

I would close this issue and maybe add a follow-up issue where we should implement or discuss the rules to add, to our config.

**Comment by bob606** [comment]
Yes I think you are right @timidak. The algorithm finds rules which are not specified config but in the plugins [...and core] I guess. So it shows you all the rules which you are not using in your config (your current-rules), but are provided in the plugins. Which would change if you explicitly turn them off.
And in fact I tested this and if you run `npm run find-unused` and take one of the rules, then switch it off explicitly in base.js, it will deisappear when you run the command another time.
